File: hw17/cmdargs.cpp
/* Example of command-line arguments in c
   Prints any command-line arguments, one per line
   Usage:  g++ -o cmdargs cmdargs.C
           cmdargs arg1 arg2 ...  */

#include <iostream>
#include <typeinfo>
using namespace std;

int main(int argc, char **argv) {
  for (int i = 0;  i < argc;  i++)
    cout << typeid(argv[i]).name() << endl;
  
  return 0;
}

File: hw22/cmdargs.py
# Example of command-line arguments in Python3
# Prints any command-line arguments, one per line
# Usage:  python3 cmdargs.py arg1 arg2 ... 

import sys
# print(sys.argv)
maxval = int(sys.argv[1])
for count, data in enumerate(sys.argv):
    if count < 2:
        continue
    current = int(data)
    maxval = max(maxval, int(data))
print(maxval)